packages/adapter-prisma/src/migrations.ts

```diff
@@ -41,6 +47,137 @@ export async function runPrototypeMigrations(dbUrl: string, schema: string, sche
+      // see the link below for what "unexecutable steps" are
+      // https://github.com/prisma/prisma-engines/blob/c65d20050f139a7917ef2efc47a977338070ea61/migration-engine/connectors/sql-migration-connector/src/sql_destructive_change_checker/unexecutable_step_check.rs
+      // the tl;dr is "making things non null when there are nulls in the db"
+      if (!migrationCanBeApplied) {
+        console.log(`${chalk.bold.red('\n⚠️ We found changes that cannot be executed:\n')}`);
+        for (const item of evaluateDataLossResult.unexecutableSteps) {
+          console.log(`  • Step ${item.stepIndex} ${item.message}`);
+        }
+      }
+      // warnings mean "if the migration was applied to the database you're connected to, you will lose x data"
+      // note that if you have a field where all of the values are null on your local db and you've removed it, you won't get a warning here.
+      // there will be a warning in a comment in the generated migration though.
+      if (evaluateDataLossResult.warnings.length) {
+        console.log(chalk.bold(`\n⚠️  Warnings:\n`));
+        for (const warning of evaluateDataLossResult.warnings) {
+          console.log(`  • ${warning.message}`);
+        }
+      }
+
+      console.log(); // for an empty line
+
+      let migrationName = await getMigrationName();
+
+      // note this only creates the migration, it does not apply it
+      let { generatedMigrationName } = await runMigrateWithDbUrl(dbUrl, () =>
+        migrate.createMigration({
+          migrationsDirectoryPath: migrate.migrationsDirectoryPath,
+          // https://github.com/prisma/prisma-engines/blob/11dfcc85d7f9b55235e31630cd87da7da3aed8cc/migration-engine/core/src/commands/create_migration.rs#L16-L17
+          // draft means "create an empty migration even if there are no changes rather than exiting"
+          // because this whole thing only happens when there are changes to the schema, this can be false
+          // (we should also ofc have a way to create an empty migration but that's a separate thing)
+          draft: false,
+          prismaSchema,
+          migrationName,
+        })
+      );
+
+      console.log(
+        `✨ A migration has been created at .keystone/prisma/migrations/${generatedMigrationName}`
+      );
+
+      let shouldApplyMigration =
+        migrationCanBeApplied && (await confirmPrompt('Would you like to apply this migration?'));
+      if (shouldApplyMigration) {
+        await runMigrateWithDbUrl(dbUrl, () => migrate.applyMigrations());
+        console.log('✅ The migration has been applied');
+      } else {
+        console.log(
+          'Please edit the migration and run keystone-next dev again to apply the migration'
+        );
+        process.exit(1);
+      }
+    } else if (!appliedMigrationNames.length) {
+      console.log('✨ Your database is already up to date, no migrations need to be applied');
+    }
```

Comment:
```suggestion
    } else {
      console.log('✨ Your database is now up to date, no further migrations need to be applied');
    }
```

Reply:
The if is intentional, why do you want to remove it? This message shouldn't be shown when there were migrations applied.

Reply:
I want to give the user a consistent experience. If we're not creating any additional migrations then they should be informed either with 
```
('✨ Your database is already up to date, no migrations need to be applied')
``` 
or
```
('✨ Your database is now up to date, no further migrations need to be applied')
```
depending on whether or not any outstanding migrations were applied during this run.

Reply:
wdyt of this?
https://github.com/keystonejs/keystone/blob/43b2f19908d0b6f18c8c07644111d24ce7985238/packages/adapter-prisma/src/migrations.ts#L153-L161